File: PythonProjects/binary_search.py
def binary_searchs(arr, target):
    low, high = 0, len(arr) - 1
    while low <= high:
        mid = (low + high) // 2
        mid_value = arr[mid]
        print(f"low {low} , mid {mid} , mid_number {mid_value}")
        if mid_value == target:
            return mid
        elif mid_value < target:
            low = mid + 1
        else:
            high = mid - 1
    return -1


sortedArray = [8,8,6,6,6,6,6,6,3,2,2,0,0,0]
target_value = 6
result = binary_searchs(sortedArray, target_value)
print(result)

File: C&CPP/algorithms/binary_search.cpp
#include<iostream>
using namespace std;

bool binarySearch(int target,int array[],int left,int right){
    while(left<=right){
        int mid=left+(right-left)/2;
        int midValue=array[mid];
       if(target==array[mid]){
            return true;
       }
       else if(target<array[mid]){
            right=mid-1;
        }else  {
            left=mid+1;
        } 
    }
    return false;
}

int main(){
    int array[] = {1, 2, 3, 4, 5};
    int left=0;
   int right=sizeof(array)/sizeof(array[0]) -1;
    bool result=binarySearch(3,array,left,right);
    if(result){
         cout<<"  found"<<endl;
    }else{
          cout<<"not found"<<endl;
    }

    return 0;
}
